nautobot/nautobot_data_validation_engine/tests/test_custom_validators.py

```diff
@@ -0,0 +1,460 @@
+            location.clean()
+
+    def test_context_processing_happy_path(self):
+        RegularExpressionValidationRule.objects.create(
+            name="Regex rule 1",
+            content_type=ContentType.objects.get_for_model(Location),
+            field="description",
+            regular_expression="{{ object.name[0:3] }}.*",
+            context_processing=True,
+        )
+
+        location = Location(
+            name="AMS-195",
+            location_type=LocationType.objects.get_by_natural_key("Region"),
+            description="AMS-195 is really cool",  # This should match `AMS.*`
+            status=Status.objects.get_by_natural_key("Active"),
+        )
+
+        try:
+            location.clean()
+        except ValidationError as e:
+            self.fail(f"rule.clean() failed validation: {e}")
+
+    def test_context_processing_sad_path(self):
+        RegularExpressionValidationRule.objects.create(
+            name="Regex rule 1",
+            content_type=ContentType.objects.get_for_model(Location),
+            field="description",
+            regular_expression="{{ object.name[0:3] }}.*",
+            context_processing=True,
+        )
+
+        location = Location(
+            name="AMS-195",
+            location_type=LocationType.objects.get_by_natural_key("Region"),
+            description="I don't like AMS-195",  # This should *not* match `AMS.*`
+            status=Status.objects.get_by_natural_key("Active"),
+        )
+
+        with self.assertRaises(ValidationError):
+            location.clean()
+
+    def test_context_processing_invalid_regex_fails_validation(self):
+        RegularExpressionValidationRule.objects.create(
+            name="Regex rule 1",
+            content_type=ContentType.objects.get_for_model(Location),
+            field="description",
+            regular_expression="[{{ object.name[0:3] }}.*",  # once processed, this is an invalid regex
+            context_processing=True,
+        )
+
+        location = Location(
+            name="AMS-195",
+            location_type=LocationType.objects.get_by_natural_key("Region"),
+            status=Status.objects.get_by_natural_key("Active"),
+        )
+
+        with self.assertRaises(ValidationError):
+            location.clean()
+
+
+class MinMaxValidationRuleModelTestCase(TestCase):
+    """
+    Test cases related to the MinMaxValidationRule model
+    """
+
+    def setUp(self) -> None:
+        wrap_model_clean_methods()
+        self.location_type = LocationType(name="Region")
+        self.location_type.validated_save()
+        return super().setUp()
+
+    def test_empty_field_values_raise_validation_error(self):
+        MinMaxValidationRule.objects.create(
+            name="Min max rule 1",
+            content_type=ContentType.objects.get_for_model(Location),
+            field="latitude",
+            min=1,
+            max=1,
+        )
+
+        location = Location(
+            name="does not match the regex",
```

Comment:
Lots of copypasta in this file - presumably this one and others in this class should be something like `name="does not satisfy the rule"` or similar?

Reply:
Change location names to reflect purposes of tests